shared/message/input/GameEvents.proto

```diff
@@ -23,84 +23,122 @@ package message.input;
         UNKNOWN_TEAM_COLOUR = 0;
         CYAN                = 1;
         MAGENTA             = 2;
     };
-
+    /// Current total score for both teams
     message Score {
-        uint32 ownScore      = 1;
+
+        /// Current total score for our team
+        uint32 ownScore = 1;
+        /// Current total score for opposition team
         uint32 opponentScore = 2;
     }
-
+    /// Goal score notification message, with total score for that team
     message GoalScored {
-        Context context    = 1;
-        uint32  totalScore = 2;
-    }
 
-    message Penalisation {
-        Context                      context = 1;
-        uint32                       robotId = 2;
-        google.protobuf.Timestamp    ends    = 3;
-        GameState.Data.PenaltyReason reason  = 4;
+        /// The actor/team who scored the goal
+        Context context = 1;
+        /// The total score of the actor/team who scored
+        uint32 totalScore = 2;
     }
+    /// Penalty notification message, with context and a resume play timestamp
+    message Penalisation {
 
-    message Unpenalisation {
+        /// The party being penalised
         Context context = 1;
-        uint32  robotId = 2;
+        /// The exact player being penalised
+        uint32 robotId = 2;
+        /// The timestamp when the robot may resume playing
+        google.protobuf.Timestamp ends = 3;
+        /// The type of infraction which caused the penalty
+        GameState.Data.PenaltyReason reason = 4;
     }
+    /// The context and robot which were unpenalised
+    message Unpenalisation {
 
-    message CoachMessage {
+        /// The context being unpenalised
         Context context = 1;
-        string  message = 2;
+        /// The specific robot being unpenalised
+        uint32 robotId = 2;
     }
+    /// Message from a coach, and the intended recipient context
+    message CoachMessage {
 
+        /// The intended recipient(s)
+        Context context = 1;
+        /// The contents of the message
+        string message = 2;
+    }
+    /// Message indicating half time has occurred
     message HalfTime {
+
+        /// True if currently first half
         bool firstHalf = 1;
     }
-
+    /// Ball kicked out message with context and timestamp
     message BallKickedOut {
-        Context                   context = 1;
-        google.protobuf.Timestamp time    = 3;
-    }
 
-    message KickOffTeam {
+        /// The party which kicked the ball out
         Context context = 1;
+        /// The time when the ball was kicked out
+        google.protobuf.Timestamp time = 3;
     }
+    /// The context of the team kicking off
+    message KickOffTeam {
 
+        /// The party kicking off
+        Context context = 1;
+    }
+    /// Set of timestamps for different parts of the game, with current phase state
     message GamePhase {
+
+        /// Current phase state, such as PLAYING or TIMEOUT
         GameState.Data.Phase phase = 1;
 
         // INITIAL
```

Comment:
what are these things?
// INITIAL, // READY, etc

Reply:
No idea

Reply:
Josephus thinks they mean that the variable below each of those comments is set after the gamecontroller has selected those respective states, so we'll run with that unless someone has any other ideas.

Reply:
This seems reasonable enough.